### DISCUSSION TITLE
Questions about thermal zone and internal heat gain

### DISCUSSION THREAD

**Discussion by Xingjian-Wang** [BIM2SIM/bim2sim#475] [Q&A] [answered]
Dear Team,

I encountered two questions when using bim2sim would like to ask. The first question is about the thermal zone, I noticed that there is a module in the documentation about adding thermal zones(_add_thermal_zones_to_building ( bldg )_ under module _bim2sim.task.bps.bind_tz_), I would like to know if it is necessary that the thermal zones should be defined in the building modeling software before generating the ifc file. The second question is about the internal heat gain of the building, I didn't find the related method in the documentation, how to add the parameter of internal heat gain of the building in the use of bim2sim?

Many thanks in advance

Regards,
Xingjian Wang

**Comment by DaJansenGit** [comment, marked as answer]
Hello Xingjan-Wang,

we just released our first version, and we are still working on the documentation, as it's not completed yet. I guess you want to simulate the building with the TEASER Plugin? As the `bind_tz` task and the `add_thermal_zones_to_building` are part of it.

**Regarding your first question:**
However, regardless of whether you want to use TEASER or EnergyPlus, the `IfcSpace`s and `IfcRelSpaceBoundary`s must be correctly present in the IFC. Please have a look at the documentation here: https://ebc.pages.rwth-aachen.de/EBC_all/github_ci/bim2sim/development/docs/overview.html#ifc-requirements 

**Regarding your second question:**
By default the internal heat gains are applied to the thermal zones based on the usage of the zone, using data from DIN18599 and SIA 2024 (data is coming from TEASER). If you want to change the loads and profiles you can do this by having a look at the[ attributes of the `ThermalZone` class ](https://ebc.pages.rwth-aachen.de/EBC_all/github_ci/bim2sim/development/docs/code/bim2sim.kernel.elements.html#bim2sim.kernel.elements.bps.ThermalZone). But documentation is still ongoing in this part of the code.

I hope this helps.
